### PR TITLE
Enable `-requireprotos` and `-warn iserror`

### PR DESCRIPTION
**Pull request by ribbanya**
Depends on #698. Prototypes are now required, and warnings are considered errors, by default. Either option can be disabled using `make` flags.

### CODE REVIEW

File: src/MSL/string.c [hunk 1]
```diff
@@ -1,5 +1,10 @@
+#include <MSL/string.h>
 #define EOF -1
+
+#ifdef MWERKS_GEKKO
```
Comment (r-burns) [inline src/MSL/string.c:4]:
```suggestion
#ifdef __MWERKS__
```
Not gekko-specific?

File: src/MSL/string.h [hunk 2]
```diff
@@ -6,9 +6,10 @@
 char* strncpy(char* __restrict dst, const char* __restrict src,
               unsigned long n);
 int __StringRead(void* isc, int ch, int Action);
-unsigned long(strlen);
 char* strchr(const char* str, int chr);
 int strncmp(const char* str1, const char* str2, unsigned long n);
 int strcmp(const char* str1, const char* str2);
+char*(strcpy) (char* __restrict dst, const char* __restrict src);
+unsigned long(strlen)(const char* s);
```
Comment (r-burns) [inline src/MSL/string.h:13]:
```suggestion
char* strcpy(char* __restrict dst, const char* __restrict src);
size_t strlen(const char* s);
```

File: src/MSL/string.c (hunk 1 above)
Comment (ribbanya) [reply src/MSL/string.c:4]: [p. 267](https://www.nxp.com/docs/en/reference-manual/CWMCUCFCMPREF.pdf)

File: src/MSL/string.h (hunk 2 above)
Comment (r-burns) [reply src/MSL/string.h:13]: Looks like got reverted somehow?

**Review by r-burns — APPROVED** [review]

**Pull request merged**